### ISSUE TITLE
Something wrong with `validate_numericality_of`?

### ISSUE THREAD

**Issue by franciscomxs** [thoughtbot/shoulda-matchers#817]
```
     Failure/Error: it { is_expected.to validate_numericality_of(:quantity) }
     Shoulda::Matchers::ActiveModel::AllowValueMatcher::CouldNotSetAttributeError:
       Expected Class to be able to set quantity to "abcd", but got 0 instead.
```

Since i'm validating numericality, the class should be expected to *not* be able to set quantity to "abcd", right? This error did appear after running `bundle update`.

**Comment by tobiasfeistmantl** [comment]
+1

**Comment by mcmire** [comment]
Yes, it's a known issue with the current release. It'll be fixed in the next release (hopefully coming soon).

**Comment by mcmire** [comment]
@franciscomxs @tobiasfeistmantl This should be fixed in 3.0.1, can you give it another try?
